frontend/public/components/instantiate-template.tsx

```diff
@@ -241,136 +220,130 @@ class TemplateForm_ extends React.Component<
-                required: requiredParam,
-                generate,
-              }: TemplateParameter) => {
-                const value = this.state.parameters[name] || '';
-                const helpID = description ? `${name}-help` : '';
-                const placeholder = generate ? t('public~(generated if empty)') : '';
-                // Only set required for parameters not generated.
-                const requiredInput = requiredParam && !generate;
-                return (
-                  <div className="form-group" key={name}>
-                    <label
-                      className={classNames('control-label', { 'co-required': requiredInput })}
-                      htmlFor={name}
-                    >
-                      {displayName || name}
-                    </label>
-                    <input
-                      type="text"
-                      className="pf-c-form-control"
-                      id={name}
-                      name={name}
-                      value={value}
-                      onChange={this.onParameterChanged}
-                      required={requiredInput}
-                      placeholder={placeholder}
-                      aria-describedby={helpID}
-                    />
-                    {description && (
-                      <div className="help-block" id={helpID}>
-                        {description}
-                      </div>
-                    )}
-                  </div>
-                );
-              },
-            )}
-            <ButtonBar
-              className="co-instantiate-template-form__button-bar"
-              errorMessage={this.state.error}
-              inProgress={this.state.inProgress}
-            >
-              <ActionGroup className="pf-c-form">
-                <Button type="submit" variant="primary">
-                  {t('public~Create')}
-                </Button>
-                <Button type="button" variant="secondary" onClick={history.goBack}>
-                  {t('public~Cancel')}
-                </Button>
-              </ActionGroup>
-            </ButtonBar>
-          </form>
-        </div>
+  return (
+    <div className="row">
+      <div className="col-md-7 col-md-push-5 co-catalog-item-info">
+        <TemplateInfo template={template} />
       </div>
-    );
-  }
-}
+      <div className="col-md-5 col-md-pull-7">
+        <form className="co-instantiate-template-form" onSubmit={save}>
+          <div className="form-group">
+            <label className="control-label co-required" htmlFor="namespace">
+              {t('public~Namespace')}
+            </label>
+            <NsDropdown selectedKey={namespace} onChange={(v) => setNamespace(v)} id="namespace" />
+          </div>
+          {params.map(
+            ({
+              name,
+              displayName,
+              description,
+              required: requiredParam,
+              generate,
+            }: TemplateParameter) => {
```

Comment:
It's generally better to type `params` when declaring the variable, instead of typing further uses of this variable.

```ts
const params: TemplateParameter[] = template.parameters || [];
```

TypeScript is capable of inferring types - if you hover the mouse over `params` declaration above and it says `params: TemplateParameter[]` without an explicit cast, there's generally no need to type it to explicitly again.